### PR TITLE
Second finalize process transfer jobs

### PR DESCRIPTION
**Pull request by diderikvw**
## Describe your changes

Brief description of your changes - not in-depth because the bulk of the description should be in the task on DevOps.

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests wherever relevant
- [ ] I have made sure that all automated checks pass before requesting a review
- [x] I do not need any deviation from our PR guidelines

### CODE REVIEW

File: services/121-service/src/seed-data/program/program-nlrc-ocw.json [hunk 1]
```diff
@@ -90,7 +90,7 @@
   "aboutProgram": {
     "en": "This is the OCW program."
   },
-  "fullnameNamingConvention": ["firstName", "lastName"],
+  "fullnameNamingConvention": ["fullName"],
```
Comment (PeterSmallenbroek) [inline services/121-service/src/seed-data/program/program-nlrc-ocw.json:93]: There should now also be a program question called fullName. During testing I found that a payment only works for PV for this specific reason.

File: services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts [hunk 2]
```diff
@@ -515,35 +515,36 @@ export class IntersolveVisaApiService {
...
+      brand: 'VISA_CARD',
+      firstName: '',
+      lastName: input.name,
+      // TODO: We need to add a "+" for Intersolve's API to work. Do we have a generic helper function for this?
+      mobileNumber: '+' + input.phoneNumber, // must match \"([+]){1}([1-9]){1}([0-9]){5,14}\"
+      cardAddress: {
+        address1:
+          `${input.addressStreet} ${input.addressHouseNumber} ${input.addressHouseNumberAddition}`.trim(),
+        city: input.addressCity,
+        country: 'NLD',
```
Comment (PeterSmallenbroek) [inline services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts:531]: Are we sure this is correct? I find it weird that for createCustomer we use 'NL' and for this one 'NLD'.

File: services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts [hunk 3]
```diff
@@ -63,25 +63,31 @@ export class TransactionJobProcessorsService {
...
-            errorMessage: errorText,
-          });
-          throw new Error(errorText);
-        }
-      }),
-    );
+
+    // Check if all required properties are present. If not, create a failed transaction and throw an error.
+    for (const key in input) {
+      if (key === 'addressHouseNumberAddition') continue; // Skip non-required property
```
Comment (PeterSmallenbroek) [inline services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts:69]: I like this early return

**Review by PeterSmallenbroek — CHANGES_REQUESTED** [review]

File: services/121-service/src/seed-data/program/program-nlrc-ocw.json (hunk 1 above)
Comment (diderikvw) [reply services/121-service/src/seed-data/program/program-nlrc-ocw.json:93]: It is there... Let's check this in a meeting, maybe I am missing something.

File: services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts (hunk 2 above)
Comment (diderikvw) [reply services/121-service/src/payments/fsp-integration/intersolve-visa/intersolve-visa.api.service.ts:531]: Yes, that is weird. And yes, this is correct. I changed it to NL and got an error back.

File: services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts (hunk 3 above)
Comment (diderikvw) [reply services/121-service/src/transaction-job-processors/transaction-job-processors.service.ts:69]: Credits go to co-pilot :)

File: services/121-service/src/seed-data/program/program-nlrc-ocw.json (hunk 1 above)
Comment (diderikvw) [reply services/121-service/src/seed-data/program/program-nlrc-ocw.json:93]: Tested with Mock Service on both OCW and PV programs, doing 2 payments in each. All works. I did find that the test data CVSs still contained first and last name. So I corrected that. Will probably conflict with Piotr's work by now :)

**Pull request merged**